lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -425,11 +427,37 @@ def update(self, instance, validated_data):
             requesting_user_id = self.context["cc_requester"]["id"]
             if key == "closed" and val:
                 instance["closing_user_id"] = requesting_user_id
+                self.trigger_thread_closed_event(instance, validated_data)
             if key == "body" and val:
                 instance["editing_user_id"] = requesting_user_id
         instance.save()
         return instance
 
+    def trigger_thread_closed_event(self, instance, validated_data):
```

Comment:
Is there a reason why this has been implemented differently to the events we have here? Why are we not using the current `tract_forum_event` method we already have, and sending in the required `data` param on it?


https://github.com/openedx/edx-platform/blob/e343dee19512606de83e3e6977c682bea8f64899/lms/djangoapps/discussion/django_comment_client/base/views.py#L68

Reply:
I was not aware of this method, and it looks promising, taking a look if we can use that.